### PR TITLE
Modules import git folder

### LINKED ISSUE #6461: import.git: support importing folders as a single module
#5547 mentions that import.git should support importing folders, but it currently only supports importing single files. import.git should be updated to support folders.

IMO, this should not include folder support in remote.git, as that would require much more thinking beyond the use case of what we need for remote.git.

Related to #6460, which is the same issue but for import.file.

### PR DESCRIPTION
**Pull request by wildum**
#### PR Description

Add support for importing directories as modules via the import.git config block.

#### Which issue(s) this PR fixes

Fixes #6461 

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

**Review by clayton-cornell — COMMENTED** [review]
Docs look OK to me

File: docs/sources/flow/reference/config-blocks/import.git.md
Comment (rfratto) [inline docs/sources/flow/reference/config-blocks/import.git.md:54]: River is a proper noun:

```suggestion
You must set the `path` attribute to a path accessible from the repository's root.
It can either be a River file such as `FILE_NAME.river` or `DIR_NAME/FILE_NAME.river` or
a directory containing River files such as `DIR_NAME` or `.` if the River files are stored at the root
of the repository.
```

File: pkg/flow/internal/controller/node_config_import.go
Comment (rfratto) [inline pkg/flow/internal/controller/node_config_import.go:1]: (I'm not going to review this, assuming that it'll be synced with the changes from #6502)

File: pkg/flow/internal/importsource/import_git.go
Comment (rfratto) [inline pkg/flow/internal/importsource/import_git.go:248]: For consistency with import.file, I'd prefer if we could add a Stat method to `im.repo` and then call `im.repo.Stat(args.Path)` here to check if the path exists and if it's a file or directory.

**Review by rfratto — APPROVED** [review]
LGTM after #6502 is merged (there's more feedback there on the shared code) and this is rebased

File: docs/sources/flow/reference/config-blocks/import.git.md
Comment (clayton-cornell) [reply docs/sources/flow/reference/config-blocks/import.git.md:54]: I missed that in the review :-(

**Pull request merged**
